pydra/engine/helpers_file.py

```diff
@@ -591,9 +591,19 @@ def template_update(inputs, output_dir, state_ind=None, map_copyfiles=None):
 
     from .specs import attr_fields
 
+    # Collect templated inputs for which all requirements are satisfied.
     fields_templ = [
-        fld for fld in attr_fields(inputs) if fld.metadata.get("output_file_template")
+        field
+        for field in attr_fields(inputs)
+        if field.metadata.get("output_file_template")
+        and all(
+            [
+                getattr(inputs, required_field) is not attr.NOTHING
+                for required_field in field.metadata.get("requires", [])
+            ]
```

Comment:
Two slight optimizations:

1) Do not construct a list. `all()` will short-circuit and avoid even finishing iterating over `field.metadata['requires']` the first time it hits a `False` value.
2) `()` is often a better empty iterable than `[]`, since tuples are immutable and the interpreter can optimize by having one global empty tuple rather than constructing a new empty list.

```suggestion
            getattr(inputs, required_field) is not attr.NOTHING
            for required_field in field.metadata.get("requires", ())
```